### ISSUE TITLE
Flip Pins on 2209?

### ISSUE THREAD

**Issue by Joebus1** [pedrsj/Creator-Pro#3]
Thank you very much for this wonderful guide.  I have klipper installed and FFCP is running better then ever.  When installing 2209 drivers do I still need to flip the pins to the other side or can I drop them in? Have the pinouts been changed via klipper?

Also, Can I set the Vref within klipper or do I need to set it using the pot?

1 More question. Do I still need to switch the two wires for each stepper or have you already done that in the cfg?

Thanks again!

**Comment by pedrsj** [comment]
You need to flip to match pins and use pot. Config file for tmc deals with direction, so no need to swap wires.

**Comment by Joebus1** [comment]
> You need to flip to match pins and use pot. Config file for tmc deals with direction, so no need to swap wires.

You are amazing! Thanks again for all your hard work. Pretty exciting getting these FFCP running better then ever.  I've got 4 of em running of an old Dell Optiplex.

**Comment by Joebus1** [comment]
Really sorry to bother you one more time Pedrsj.  I'm trying to flash another creator pro here at my house.  I keep getting this error.
I'm using a Inovato Quadra with klipper installed etc.   I'm able to `ls` the serial from the creator pro so I know it's not the usb cable.

  `quadra@FFCP:~/klipper$ avrdude -v -c stk500v2 -p atmega2560 -P /dev/serial/by-id/usb-MakerBot_Industries_The_Replicator_55736303939351209060-if0 -b 115200 -D -U out/klipper.elf.hex

avrdude: Version 6.3-20171130
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "/etc/avrdude.conf"
         User configuration file is "/home/quadra/.avrduderc"
         User configuration file does not exist or is not a regular file, skipping

         Using Port                    : /dev/serial/by-id/usb-MakerBot_Industries_The_Replicator_55736303939351209060-if0
         Using Programmer              : stk500v2
         Overriding Baud Rate          : 115200
avrdude: ser_open(): can't open device "/dev/serial/by-id/usb-MakerBot_Industries_The_Replicator_55736303939351209060-if0": No such file or directory

avrdude done.  Thank you.
` 
I'm going to try using a pi tonight as I think it has to do with com port assignments in the quadra.  I don't know how to assign them or do what I need to.  Thanks again for all your knowledge!

**Comment by Joebus1** [comment]
[klippy.log](https://github.com/pedrsj/Creator-Pro/files/11504175/klippy.log)
Sorry to bug you again.  I'm running into this issue when I run your `START_PRINT` I get the error `Move out of range: 0.000 -16.193 0.250 [88.000]` I have the Extruder 1 selected(Left side). It happens right after the purge line takes place. I'm not sure what I've got wrong. Only thing I can think of is something to do with Origin?  I'm still learning so no idea. Thank you.

**Comment by pedrsj** [comment]
No need to apologise for asking for help, I just may take a while to respond as work is a little insane at the moment.

The origin is left corner at front of machine, which is different to the stock firmware origin of the centre of the bed. I did it this way to be consistent with all my other printers as they share macros.

As it is after a successful purge, the error is occurring as it starts the actual print, so this would indicate that the movement mode G90 (absolute) / G91 (relative) is being set to relative i.e. G91, the start macro sets it to G90 (absolute).

What slicer are you using and are you defining your own GCode?

**Comment by Joebus1** [comment]
I'm using Superslicer. Also I am using your configs copy pasted along with your `START_PRINT` script.  Origin was set to half of max X and Y. I'm going to change to 0,0 and try again.  Do you happen to have a superslicer profile to share so I can compare some things?  Thanks again so much. This is awesome getting klipper on these.

**Comment by pedrsj** [comment]
That would also explain it, I'll update the instructions.

**Comment by pedrsj** [comment]
If you copied everything in the instructions the only difference would be the origin values.

**Comment by Joebus1** [comment]
Yay, Moving origin to 0,0 in the slicer fixed the issue. Only issue I'm seeing now is unless I activate the Left extruder(extruder1) manually via the macro in mainsail, The printer will heat up the Right extruder(extruder0) but not the left one. I don't remember this happening before so I must of changed something somewhere(swear I didn't haha)  The attachment shows where I choose the extruder I want to use in SS. Is this correct?   
![image](https://github.com/pedrsj/Creator-Pro/assets/106839320/5bd3554a-7f80-4bf8-952a-8be7c2f155ed)
I'm not sure how to set these either.
![image](https://github.com/pedrsj/Creator-Pro/assets/106839320/7a72e0b9-d887-474b-8415-da3a477181f2)

**Comment by pedrsj** [comment]
If I understand your question correctly, you are only wanting to print with a single extruder? If so, you need to set both extruders to be the filament you are going to print. With the available slicer variables, I can only determine the number of extruders in use by comparing the selected filament e.g. 1 if both extruder filaments are the same, otherwise 2. At the end of the instructions it shows this with a screen image. You then just select the actual extruder you want to print your model with.

I do the selection this way to avoid changing start gcode if you swap extruders.

**Comment by pedrsj** [comment]
The assumption of course is that you will always be using different filaments / colours when using both extruders.


START_PRINT BED=[first_layer_bed_temperature] 
EXTRUDER={first_layer_temperature[0]+extruder_temperature_offset[0]} 
EXTRUDER1={first_layer_temperature[1]+extruder_temperature_offset[1]} 
MESH=false 
EXTRUDERS_ENABLED={(filament_preset[0] != filament_preset[1]? 2 : 1)} # determine number of extruders by comparing if the filaments are the same
INITIAL_EXTRUDER=[initial_extruder]  # extruder you start printing with

**Comment by Joebus1** [comment]
Sorry for the long post:

I understand now. :)  I believe I mostly wanted to know if my slicer settings were correct, but I think that is more for the SuperSlicer forums.  

 The funny behavior I experienced was when I wanted to print with just the left Extruder only. Before I fixed the origin problem I would select the left extruder in superslicer. Then slice and send it to mainsail and it would start heating up the correct extruder.  After I fixed the origin problem I followed the same steps but the wrong extruder was heating up now. Then klipper would give an error about the correct extruder being under temp.  To make it work I have to click the "T1" macro button in mainsail. Then start the print and it heats up and prints correctly. I would not think changing origin would effect that behavior so I think it  must be something I did by accident somewhere.

![image](https://github.com/pedrsj/Creator-Pro/assets/106839320/e19460b5-f75c-49f8-b3e6-28bf0f4b79a7)

 I don't know what the heck I was going on about in the previous post. It was really late in the night and I am really bad at explaining things sometimes. Reading my own post and even I'm not sure what I'm asking. :)  Thanks again for taking the time on the guide and helping me here.  I hope to swap for a SKR mini E3 at some point soon.

**Comment by pedrsj** [comment]
Do you still have to manually select an extruder?

On changing the board be mindful that the creator pro uses type k thermocouples and not thermistors, you'd need extra hardware to use the existing hotends with most 3d printer boards.

**Comment by Joebus1** [comment]
> Do you still have to manually select an extruder?

No, It is working as it should now. I did a full power cycle of everything and copied over the config again, not sure what that was.  I'm running the first full print now.

> On changing the board be mindful that the creator pro uses type k thermocouples and not thermistors, you'd need extra hardware to use the existing hotends with most 3d printer boards.

Thank you. I found this [https://msptrick.blogspot.com/2022/11/flashforge-creator-pro-control-board.html](url) and he happens to mention that :)

**Comment by pedrsj** [comment]
Other than you can, what  is the reason to change the board

**Comment by Joebus1** [comment]
I figure I will spend half the price of the board on drivers anyways, then I got to spend time to flip the pins, The mighty board is a good dependable board but I assume newer might be better.  

Mostly because I want to try ;)

**Comment by pedrsj** [comment]
Will be a cool project to do.
